### PR TITLE
Add notes on how to implement custom DateTime(Offset) processing with JSON reader/writer

### PR DESCRIPTION
**Pull request by layomia**
Addresses https://github.com/dotnet/docs/pull/13526#discussion_r314139635.

### CODE REVIEW

File: docs/standard/datetime/system-text-json-support.md [hunk 1]
```diff
@@ -110,6 +113,28 @@ the converter successfully parses the data using <xref:System.DateTime.Parse(Sys
...
+
+[!code-csharp[example-custom-writing-with-utf8jsonwriter](~/samples/snippets/standard/datetime/json/custom-writing-with-utf8jsonwriter.cs)]
+
+### When reading with <xref:System.Text.Json.Utf8JsonReader>
+
+If you want to read a custom <xref:System.DateTime> or <xref:System.DateTimeOffset> text representation with <xref:System.Text.Json.Utf8JsonReader>,
+you can get the value of the current JSON token as a <xref:System.String> using <xref:System.Text.Json.Utf8JsonReader.GetString>, then parse the value
+using custom logic.
+
+The following example shows how a custom <xref:System.DateTimeOffset> text representation can be retrieved using <xref:System.Text.Json.Utf8JsonReader.GetString>,
```
Comment (ahsonkhan) [inline docs/standard/datetime/system-text-json-support.md]: Other than showcasing the exception path first, before the workaround, this looks good:
https://github.com/dotnet/samples/pull/1351/files#r319257875

Comment (layomia) [reply docs/standard/datetime/system-text-json-support.md]: @ahsonkhan I've addressed this in https://github.com/dotnet/samples/pull/1412 and added a commit to this PR, please take a look.

Comment (ahsonkhan) [reply docs/standard/datetime/system-text-json-support.md]: Re-looking at the whole doc, the latest changes and dotnet/samples#1412 may be redundant and hence we could undo them.

**Review by ahsonkhan — APPROVED** [review]

File: docs/standard/datetime/system-text-json-support.md [hunk 2]
```diff
@@ -194,4 +219,6 @@ The following levels of granularity are defined for formatting:
 
     4. "yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'FFFFFFF('+'/'-')HH':'mm"
 
-        Used to format a <xref:System.DateTime> or <xref:System.DateTimeOffset> with fractional seconds and with a local offset.
+		Used to format a <xref:System.DateTime> or <xref:System.DateTimeOffset> with fractional seconds and with a local offset.
```
Comment (ahsonkhan) [inline docs/standard/datetime/system-text-json-support.md]: Why is this indented so deep now?

**Comment by layomia** [comment]
@rpetrusha, please can you take a look?

**Comment by ahsonkhan** [comment]
Can we merge https://github.com/dotnet/samples/pull/1418, so that this doc update can be merged as well? Look's like it is ready.

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build after manual publish to sync with Samples reop.

**Review by rpetrusha — APPROVED** [review]
This looks good, @layomia, and the build is now successful. I'll merge.

**Pull request merged**